packages/subnav/index.js

```diff
@@ -2,6 +2,7 @@ import classNames from 'classnames'
 import useProductMeta from '@hashicorp/platform-product-meta'
 import useNavRef from './helpers/useNavRef'
 import s from './style.module.css'
+import themeStyles from './theme.module.css'
```

Comment:
Took a theming approach similar to that in `code-block`. While I am certainly not confident this will be the best approach to theming long-term (in fact I'm almost sure we can come up with something more extensible for product situations), this felt like a decent balance between using our existing tooling and conventions (CSS modules & custom properties) while splitting out theme properties.

My hot take is I don't think the theming is something we should aim to "solve" with this PR, really just trying to get this to work and be easy to transition to whatever theming approach we go with longer term. Very open to suggestions here, of course 🙇